src/test/java/org/jabref/logic/util/io/RegExpBasedFileFinderTests.java

```diff
@@ -80,6 +80,19 @@ void testYearAuthFirstPageFindFiles() throws Exception {
                 result);
     }
 
+    @Test
+    void findAssociatedFilesContainingRegexpFromBracketedExpression() throws Exception {
+        var bibEntry = new BibEntry().withField(StandardField.TITLE, "Regexp from [A-Z]");
+
+        var extension = Collections.singletonList("pdf");
+        var directory = Collections.singletonList(Path.of(FILES_DIRECTORY));
+        RegExpBasedFileFinder fileFinder = new RegExpBasedFileFinder("[TITLE]\\\\.[extension]", ',');
+
+        List<Path> result = fileFinder.findAssociatedFiles(bibEntry, directory, extension);
+        assertEquals(Collections.singletonList(Path.of("src/test/resources/org/jabref/logic/importer/unlinkedFilesTestFolder/Regexp from [A-Z].pdf")),
+                result);
```

Comment:
Test for bracketed pattern

Reply:
`GUO ea - INORG CHEM COMMUN 2010 - Ferroelectric Metal Organic Framework (MOF).pdf` should be a problem in the current master. The example is from the zip file in https://github.com/JabRef/jabref/issues/4342#issuecomment-686606696.

Reply:
Why is that an issue? - Could you simply add a test case for that filename, too? Better more tests even if they cover the same lines of code. If someone else touches the code later, it might help.

Reply:
> Could you simply add a test case for that filename, too?

Will do.

> Why is that an issue?

Because the RegExpBasedFileFinder interpret(ed) the content of an expanded bracketed pattern as if it is part of a regex. In this case, the title contains `(MOF)` which in regex is a capturing group matching `MOF`, not `(MOF)`.